analysis/behavioral_analysis/non_imaging_analysis_nback_group_comparison.ipynb

```diff
@@ -0,0 +1,549 @@
+    "import numpy as np\n",
+    "import re\n",
+    "import glob\n",
+    "import json\n",
+    "import matplotlib.pyplot as plt\n",
+    "import statsmodels\n",
+    "from scipy import stats\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "id": "52813f52",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#prepare data from MMJ-Processed_data-2022_05_27-13_58-6858bbe.csv\n",
+    "\n",
+    "\n",
+    "def create_subs_df(group,ses,task,contrast):\n",
+    "    effect_size_maps = glob.glob(f'../../../derivatives/task_analysis_volume/first_level/sub-{group}*/ses-{ses}/task-{task}/sub-{group}*_ses-{ses}_task-{task}_rec-unco_run-1_contrast-{contrast}_effect_size.nii.gz')\n",
+    "    non_img_data = pd.read_csv(f\"../../../sourcedata/non_imaging_data/MMJ-Processed_data-2022_05_27-13_58-6858bbe.csv\",low_memory=False)\n",
+    "\n",
+    "    #need to find subjects that have data for this contrast\n",
+    "    subs = [path.split('/sub-')[1].split('/')[0] for path in effect_size_maps if path]\n",
+    "    subs = ['_'.join([s for s in re.split(r'(MM|HC)', sub) if s]) for sub in subs]\n",
+    "    df_subs = pd.DataFrame(subs,columns=['subs'])\n",
+    "\n",
+    "\n",
+    "    #add columns for male and female, that will then be combined to create the group average \n",
+    "    grouped_sex = non_img_data.groupby(\"IDS.CHR.Subject\")[\"SBJ.CHR.Sex\"].agg(\"first\")\n",
+    "    dict_sex = grouped_sex.to_dict()\n",
+    "    df_subs = pd.concat([df_subs,pd.get_dummies(df_subs['subs'].map(dict_sex))],axis=1,copy=False)\n",
+    "\n",
+    "    #add age, mean-centered\n",
+    "    grouped_age = non_img_data.groupby(\"IDS.CHR.Subject\")[\"SBJ.INT.Age\"].agg(\"first\")\n",
+    "    dict_age = grouped_age.to_dict()\n",
+    "    df_subs['age'] = df_subs['subs'].map(dict_age)\n",
+    "    df_subs['age'] = df_subs['age'] - df_subs['age'].mean()\n",
+    "\n",
+    "    #add CUDIT summed score, mean-centered\n",
+    "    if group == 'HC': \n",
+    "        grouped_HC_baseline_cudit = non_img_data[non_img_data['SSS.CHR.Time_point'] == 'Screening'].groupby('IDS.CHR.Subject')['INV.INT.CUDIT.Summed_score'].agg(\"first\")\n",
+    "        dict_HC_baseline_cudit = grouped_HC_baseline_cudit.to_dict()\n",
+    "        df_subs['total_cudit'] = df_subs['subs'].map(dict_HC_baseline_cudit)\n",
+    "        df_subs['total_cudit'] = df_subs['total_cudit'] - df_subs['total_cudit'].mean()\n",
+    "\n",
+    "    else:\n",
+    "        if ses == 'baseline':\n",
+    "            dict_MM_baseline_cudit = non_img_data[non_img_data['SSS.CHR.Time_point'] == 'Baseline'].groupby('IDS.CHR.Subject')['INV.INT.CUDIT.Summed_score'].agg(\"first\").to_dict()\n",
+    "            df_subs['total_cudit'] = df_subs['subs'].map(dict_MM_baseline_cudit)\n",
+    "            df_subs['total_cudit'] = df_subs['total_cudit'] - df_subs['total_cudit'].mean()\n",
+    "            df_subs['total_cudit'].fillna(0, inplace=True)\n",
```

Comment:
this does mean that you are saying the people missing have the same as the mean cudit. in general, this should either be imputed or the subjects with missing data removed.

Reply:
one way to remove these subjects is to add columns with a 1 for each subject with a missing value. similar to how at the first level outlier scans are omitted.